pkg/semtech-loramac/contrib/semtech_loramac_radio.c

```diff
@@ -0,0 +1,256 @@
+    sx127x_set_rx_single(&sx127x, !rxContinuous);
+    sx127x_set_rx(&sx127x);
+}
+
+void SX1276SetTxConfig(RadioModems_t modem, int8_t power, uint32_t fdev,
+                       uint32_t bandwidth, uint32_t spreading_factor,
+                       uint8_t coding_rate, uint16_t preambleLen,
+                       bool fixLen, bool crcOn, bool freqHopOn,
+                       uint8_t hopPeriod, bool iqInverted, uint32_t timeout)
+{
+    (void) fdev;
+    (void) fixLen;
+    sx127x_set_modem(&sx127x, modem);
+    sx127x_set_freq_hop(&sx127x, freqHopOn);
+    sx127x_set_bandwidth(&sx127x, bandwidth);
+    sx127x_set_coding_rate(&sx127x, coding_rate);
+    sx127x_set_spreading_factor(&sx127x, spreading_factor);
+    sx127x_set_crc(&sx127x, crcOn);
+    sx127x_set_freq_hop(&sx127x, freqHopOn);
+    sx127x_set_hop_period(&sx127x, hopPeriod);
+    sx127x_set_fixed_header_len_mode(&sx127x, false);
+    sx127x_set_iq_invert(&sx127x, iqInverted);
+    sx127x_set_payload_length(&sx127x, 0);
+    sx127x_set_tx_power(&sx127x, power);
+    sx127x_set_preamble_length(&sx127x, preambleLen);
+    sx127x_set_rx_single(&sx127x, false);
+    sx127x_set_tx_timeout(&sx127x, timeout * US_PER_MS); /* base unit us, LoRaMAC ms */
+}
+
+uint32_t SX1276GetTimeOnAir(RadioModems_t modem, uint8_t pktLen)
+{
+    (void) modem;
+    return sx127x_get_time_on_air(&sx127x, pktLen);
+}
+
+void SX1276Send(uint8_t *buffer, uint8_t size)
+{
+    netdev_t *dev = (netdev_t *)&sx127x;
+    struct iovec vec[1];
+    vec[0].iov_base = buffer;
+    vec[0].iov_len = size;
+    dev->driver->send(dev, vec, 1);
+}
+
+void SX1276SetSleep(void)
+{
+    sx127x_set_sleep(&sx127x);
+}
+
+void SX1276SetStby(void)
+{
+    sx127x_set_standby(&sx127x);
+}
+
+void SX1276SetRx(uint32_t timeout)
+{
+    (void) timeout;
+    sx127x_set_rx(&sx127x);
+}
+
+void SX1276StartCad(void)
+{
+    sx127x_start_cad(&sx127x);
+}
+
+int16_t SX1276ReadRssi(RadioModems_t modem)
+{
+    sx127x_set_modem(&sx127x, (uint8_t)modem);
+    return sx127x_read_rssi(&sx127x);
+}
+
+void SX1276Write(uint8_t addr, uint8_t data)
+{
+    sx127x_reg_write(&sx127x, addr, data);
+}
+
+uint8_t SX1276Read(uint8_t addr)
+{
+    return sx127x_reg_read(&sx127x, addr);
+}
+
+void SX1276WriteBuffer(uint8_t addr, uint8_t *buffer, uint8_t size)
+{
+    sx127x_reg_write_burst(&sx127x, addr, buffer, size);
+}
+
+void SX1276ReadBuffer(uint8_t addr, uint8_t *buffer, uint8_t size)
+{
+    sx127x_reg_read_burst(&sx127x, addr, buffer, size);
+}
+
+void SX1276SetMaxPayloadLength(RadioModems_t modem, uint8_t max)
+{
+    (void) modem;
+    sx127x_set_max_payload_len(&sx127x, max);
+}
+
+bool SX1276CheckRfFrequency(uint32_t frequency)
+{
+    (void) frequency;
+    /* Implement check. Currently all frequencies are supported */
+    return true;
+}
+
+void SX1276SetTxContinuousWave(uint32_t freq, int8_t power, uint16_t time)
+{
+    (void) freq;
+    (void) power;
+    (void) time;
+    /* TODO */
```

Comment:
TODO ?

Reply:
Yes, if you look at the upstream code ([sx1276](https://github.com/Lora-net/LoRaMac-node/blob/master/src/radio/sx1276/sx1276.c#L1124), [sx1272](https://github.com/Lora-net/LoRaMac-node/blob/master/src/radio/sx1272/sx1272.c#L972)), this is something that should be in the driver, and we don't support that for the moment. It also switch the radio modem to FSK mode and this mode is not supported by RIOT's driver. I don't know what are the implications.